### PR TITLE
cleans up warning/errors tensorflow/stream_executor 

### PR DESCRIPTION
**Pull request by Mistobaan**
- the delete variables are not used anywhere (why keep them around?)
- tests seem to be running as before.

### CODE REVIEW

File: tensorflow/stream_executor/cuda/cuda_activation.cc [hunk 1]
```diff
@@ -27,8 +27,7 @@ CudaContext* ExtractCudaContext(CUDAExecutor *cuda_exec);
 CUDAExecutor *ExtractCudaExecutor(StreamExecutor *stream_exec);
 
 ScopedActivateExecutorContext::ScopedActivateExecutorContext(
-    CUDAExecutor *cuda_exec)
-    : cuda_exec_(cuda_exec),
+    CUDAExecutor *cuda_exec):
```
Comment (vrv) [inline tensorflow/stream_executor/cuda/cuda_activation.cc]: I don't think this is what clang-format would do, revert?

Comment (vrv) [inline tensorflow/stream_executor/cuda/cuda_activation.cc]: Actually I'm not sure why you made this change at all...

File: tensorflow/stream_executor/cuda/cuda_dnn.cc [hunk 2]
```diff
@@ -457,6 +457,7 @@ class ScopedFilterDescriptor {
                  << ToString(status);
     }
 
+#if CUDNN_VERSION >= 5000
```
Comment (vrv) [inline tensorflow/stream_executor/cuda/cuda_dnn.cc]: Why is this needed?  We build with cudnnv4 right now and nothing was failing...

File: tensorflow/stream_executor/cuda/cuda_dnn.cc [hunk 3]
```diff
@@ -666,7 +668,7 @@ class ScopedActivationDescriptor {
         mode = CUDNN_ACTIVATION_TANH;
         break;
       default:
-        LOG(ERROR) << "unrecognized activation mode: "
+        LOG(FATAL) << "unrecognized activation mode: "
```
Comment (vrv) [inline tensorflow/stream_executor/cuda/cuda_dnn.cc]: Let's revert this -- we don't want to crash the program -- we'd prefer to return a bad status message.

File: tensorflow/stream_executor/kernel.h [hunk 4]
```diff
@@ -349,6 +349,7 @@ class TypedKernel : public KernelBase {
         port::InlinedVector<uint8, 4>{arg_ptr, arg_ptr + sizeof(arg)}, sizeof(arg)});
   }
 
+#ifndef GOOGLE_CUDA
```
Comment (vrv) [inline tensorflow/stream_executor/kernel.h]: this looks wrong -- or rather, there's a bug somewhere else that needs to be fixed, rather than adding this here

File: tensorflow/stream_executor/cuda/cuda_dnn.cc (hunk 2 above)
Comment (Mistobaan) [inline tensorflow/stream_executor/cuda/cuda_dnn.cc]: otherwise for CUDNNv4 and less  the `format` variable will be defined but not used (see the other block below:
```c++
    status = dynload::cudnnSetFilterNdDescriptor(parent_, handle_, elem_type,
#if CUDNN_VERSION >= 5000
                                                 format,
#endif
                                                 dims.size(), dims.data());
```

File: tensorflow/stream_executor/cuda/cuda_dnn.cc (hunk 3 above)
Comment (Mistobaan) [inline tensorflow/stream_executor/cuda/cuda_dnn.cc]: ok

File: tensorflow/stream_executor/kernel.h (hunk 4 above)
Comment (Mistobaan) [inline tensorflow/stream_executor/kernel.h]: removed

File: tensorflow/stream_executor/cuda/cuda_activation.cc (hunk 1 above)
Comment (Mistobaan) [inline tensorflow/stream_executor/cuda/cuda_activation.cc]: @vrv I think this is suspicious, not sure who owns the stream_executor but could it be that because we are not deleting the cuda_exec once it scopes out, will cause #1727 ? I did not have time to dig in the code to find out .

Comment (vrv) [inline tensorflow/stream_executor/cuda/cuda_activation.cc]: I don't think this object owns the cuda_exec pointer, so it's probably fine.  I guess it was just unused.

File: tensorflow/stream_executor/cuda/cuda_dnn.cc (hunk 3 above)
Comment (vrv) [inline tensorflow/stream_executor/cuda/cuda_dnn.cc]: Good point, this class is not written very well.  One day we should probably refactor this so that we can check the status rather than just kill the program.

File: tensorflow/python/ops/rnn.py [hunk 5]
```diff
@@ -126,7 +126,8 @@ def rnn(cell, inputs, initial_state=None, dtype=None,
       state = initial_state
     else:
       if not dtype:
-        raise ValueError("If no initial_state is provided, dtype must be.")
+        raise ValueError("If no initial_state is provided, "
+                           "dtype must be specified")
```
Comment (vrv) [inline tensorflow/python/ops/rnn.py]: alignment is messed up here -- this seems unrelated to this commit, maybe put in a different PR?

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Comment by martinwicke** [comment]
Looks ready to merge, vrv@?
